### ISSUE TITLE
Incompatibility of 3P auto-complete with BetterMultiSelection

### ISSUE THREAD

**Issue by pidgeon777** [jcaillon/3P#247]
BetterMultiSelection is a great Notepad++ Plugin to improve the default multi-selection:

https://github.com/dail8859/BetterMultiSelection

The problem arises when both 3P auto-complete and BetterMultiSelection are active:

When trying to select the previous or next word in the auto-complete suggestion list by pressing the Down or Up keyboard arrows, BetterMultiSelection seems to intervene instead of 3P:

![yq1zyyuytc](https://user-images.githubusercontent.com/13739892/50076945-4ea2b200-01e3-11e9-976c-670bd75e63d8.gif)

In the example posted above, pressing Down/Up arrows when the auto-complete box is active has the effect of moving all of the active cursors Up/Down (this is the behavior of BetterMultiSelection), instead of correctly selecting the previous/next word suggested in the auto-complete list.

Software/OS:

Notepad++ v7.5.9 32bit
3P - Progress Programmers Pal v1.8.3
Windows 10 x64

Thank you

**Comment by jcaillon** [comment]
Hi,

thank you for the crystal clear issue. 

Unfortunately I can't do anything in that particular situation. Looking at the other plugin code, we use the same windows hook to react on keyboard events but the other plugin is called first, handles the UP/DOWN arrow key and does not propagate the event to other subscriber (i.e. 3P) since it has handled the event. Thus 3P never even receives the UP/DOWN key being pressed.

Bottom line is, there is nothing I can do on my side to fix the compatibility issue. The other plugin could stop handling the event when detecting 3P autocompletion window but I doubt it is something the maintainer would want to do. Another solution is to implement the shortcuts of the other plugin in 3P. I will keep this issue open with that in mind. Be aware this can take a loooot of time though :/

Sorry

**Comment by pidgeon777** [comment]
Thanks for your reply. First of all, here is the tool I used for recording the action:

https://getsharex.com/

It is a great and light open source software which also includes a "Record video region to Gif" through which I recorded the sequence posted above.

Regarding the bug I reported, there is some news: while playing around with both BetterMultiSelection and 3P plugins, I discovered that when the autocomplete box is active (with multiple selections), to select the next/previous suggested item of the list it is sufficient to activate one the following keyboard combinations:
- **Alt** + **Up/Down arrows**.
- **Ctrl** + **Up/Down arrows**.

By pressing the Alt/Ctrl keyboard modifiers, the Up/Down actions are correctly handled by 3P autocomplete function, temporarily disabling the upward/downward movements of the multiple selections.

Moreover, if I simply press the Enter key to select a suggested item when multiple selections are active, then the enter Key simply propagates for each selection adding a newline, discarding the suggested item. To actually append the selected suggested item even when multiple selections are active, it is enough to press one of the following sequences:
- **Alt** + **Enter**
- **Ctrl** + **Enter**

or simply **Tab**, which would also work.

So, at the moment I think this could do the trick, giving me a chance to use both 3P autocomplete function and BetterMultiSelection plugin at the same time, with satisfying results.
